ledger/participant-integration-api/src/main/scala/platform/store/dao/events/BufferedTransactionsReader.scala

```diff
@@ -109,60 +121,104 @@ private[events] class BufferedTransactionsReader(
 private[platform] object BufferedTransactionsReader {
   def apply(
       delegate: LedgerDaoTransactionsReader,
-      transactionsBuffer: EventsBuffer[TransactionLogUpdate],
+      transactionsBuffer: InMemoryFanoutBuffer,
       eventProcessingParallelism: Int,
       lfValueTranslation: LfValueTranslation,
       metrics: Metrics,
   )(implicit
       executionContext: ExecutionContext
   ): BufferedTransactionsReader = {
-    val fetchFlatTransactionsFromPersistence =
-      new FetchFromPersistence[(FilterRelation, Boolean), GetTransactionsResponse] {
-        override def apply(
-            startExclusive: Offset,
-            endInclusive: Offset,
-            filter: (FilterRelation, Boolean),
-        )(implicit
-            loggingContext: LoggingContext
-        ): Source[(Offset, GetTransactionsResponse), NotUsed] = {
-          val (filterRelation, verbose) = filter
-          delegate.getFlatTransactions(startExclusive, endInclusive, filterRelation, verbose)
-        }
-      }
-    val fetchTransactionTreesFromPersistence =
-      new FetchFromPersistence[(Set[Party], Boolean), GetTransactionTreesResponse] {
-        override def apply(
-            startExclusive: Offset,
-            endInclusive: Offset,
-            filter: (Set[Party], Boolean),
-        )(implicit
-            loggingContext: LoggingContext
-        ): Source[(Offset, GetTransactionTreesResponse), NotUsed] = {
-          val (requestingParties, verbose) = filter
-          delegate.getTransactionTrees(startExclusive, endInclusive, requestingParties, verbose)
-        }
-      }
     val flatTransactionsStreamReader =
       new BufferedStreamsReader[(FilterRelation, Boolean), GetTransactionsResponse](
-        transactionLogUpdateBuffer = transactionsBuffer,
-        fetchFromPersistence = fetchFlatTransactionsFromPersistence,
+        inMemoryFanoutBuffer = transactionsBuffer,
+        fetchFromPersistence =
+          new FetchFromPersistence[(FilterRelation, Boolean), GetTransactionsResponse] {
+            override def apply(
+                startExclusive: Offset,
+                endInclusive: Offset,
+                filter: (FilterRelation, Boolean),
+            )(implicit
+                loggingContext: LoggingContext
+            ): Source[(Offset, GetTransactionsResponse), NotUsed] = {
+              val (filterRelation, verbose) = filter
+              delegate.getFlatTransactions(startExclusive, endInclusive, filterRelation, verbose)
+            }
+          },
         bufferedStreamEventsProcessingParallelism = eventProcessingParallelism,
         metrics = metrics,
         streamName = "transactions",
       )
+
     val transactionTreesStreamReader =
       new BufferedStreamsReader[(Set[Party], Boolean), GetTransactionTreesResponse](
```

Comment:
One difference between BufferedStreamsReader and BufferedTransactionByIdReader is that the latter gets the toApi method passed into the constructor the other it is at the method level.  Could both pass in at the constructor level ?

Reply:
This is only for convenience and reducing verbosity. I'll change if you feel strongly about it.